src/rgw/rgw_lc.cc

```diff
@@ -397,15 +436,15 @@ int RGWLC::bucket_lc_process(string& shard_id)
             is_expired = obj_has_expired(now - ceph::real_clock::to_time_t(obj_iter->meta.mtime), prefix_iter->second.expiration);
           }
           if (is_expired) {
-            RGWObjectCtx rctx(store);
-            rgw_obj obj(bucket_info.bucket, key);
-            RGWObjState *state;
             int ret = store->get_obj_state(&rctx, bucket_info, obj, &state, false);
             if (ret < 0) {
               return ret;
             }
-            if (state->mtime != obj_iter->meta.mtime)//Check mtime again to avoid delete a recently update object as much as possible
+            if (state->mtime != obj_iter->meta.mtime) {
```

Comment:
Currently this seems to be valid if an object is modified via say a put_object_tagging operation, in which case the mtime from state is older than the one in bucket index. And this will end up with a case of the object never getting deleted

Reply:
For eg: 
```
skipping removal: state->mtime 2017-08-28 16:17:44.0.565582s obj->mtime 2017-08-28 16:17:44.0.574707s
```